vscode/src/ruby/versionManager.ts

```diff
@@ -59,28 +58,36 @@ export abstract class VersionManager {
   // language server
   abstract activate(): Promise<ActivationResult>;
 
-  protected async runEnvActivationScript(activatedRuby: string) {
+  protected async runEnvActivationScript(
+    activatedRuby: string,
+  ): Promise<ActivationResult> {
+    const activationUri = vscode.Uri.joinPath(
+      this.context.extensionUri,
+      "activation.rb",
+    );
+
     const result = await this.runScript(
-      `${activatedRuby} -W0 -rjson -e '${this.activationScript}'`,
+      `${activatedRuby} '${activationUri.fsPath}'`,
     );
 
     const activationContent = new RegExp(
-      `${ACTIVATION_SEPARATOR}(.*)${ACTIVATION_SEPARATOR}`,
+      `${ACTIVATION_SEPARATOR}([^]*)${ACTIVATION_SEPARATOR}`,
     ).exec(result.stderr);
 
-    return this.parseWithErrorHandling(activationContent![1]);
-  }
+    const [version, gemPath, yjit, ...env] =
+      activationContent![1].split(VALUE_SEPARATOR);
 
-  protected parseWithErrorHandling(json: string) {
-    try {
-      return JSON.parse(json);
-    } catch (error: any) {
-      this.outputChannel.error(
-        `Tried parsing invalid JSON environment: ${json}`,
-      );
-
-      throw error;
+    const groupedEnv = [];
+    for (let i = 0; i < env.length; i += 2) {
+      groupedEnv.push([env[i], env[i + 1]]);
     }
```

Comment:
If we are building something anyway, I think we can build the target object directly here:
```suggestion
    const groupedEnv = {};
    for (let i = 0; i < env.length; i += 2) {
      groupedEnv[env[i]] = env[i + 1];
    }
```